sources/packages/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -273,4 +287,124 @@ export class AssessmentControllerService {
       });
     return finalAward;
   }
+
+  /**
+   * Get all requests related to an assessments for a student
+   * application, i.e, this will fetch all pending and denied
+   * student appeals for an application or possible application
+   * exceptions that will prevent the assessment to proceed till
+   * they are approved, for instance, when a document is uploaded
+   * and need to be reviewed.
+   * @param applicationId application number.
+   * @param studentId applicant student.
+   * @param  includeOnlyAppeals then only student appeals are returned and
+   * other assessment types must not be available when this property is true.
+   * @returns assessment requests or exceptions for a student application.
+   */
+
+  async getRequestedAssessmentSummary(
+    applicationId: number,
+    studentId?: number,
+    includeOnlyAppeals = false,
+  ): Promise<RequestAssessmentSummaryAPIOutDTO[]> {
+    const requestAssessmentSummary: RequestAssessmentSummaryAPIOutDTO[] = [];
+    // Requested offering changes and application exceptions must not be included when includeOnlyAppeals is true.
+    if (!includeOnlyAppeals) {
```

Comment:
It seems the queries inside this "if" are executed without checking if the student has access to the application provided as a parameter, is it right? If it is, we should either check the authorization before the queries are executed or include the authorization in the query (for students only).